pkg/sql/tenant.go

```diff
@@ -698,3 +698,33 @@ WHERE id = $1`, tenantID, tenantName); err != nil {
 
 	return nil
 }
+
+// GetTenantInfo implements the tree.TenantOperator interface.
+func (p *planner) GetTenantInfo(
+	ctx context.Context, tenantName roachpb.TenantName,
+) (*descpb.TenantInfo, error) {
+	if err := p.RequireAdminRole(ctx, "get tenant"); err != nil {
+		return nil, err
+	}
+
+	if !p.EvalContext().Settings.Version.IsActive(ctx, clusterversion.V23_1TenantNames) {
+		return nil, pgerror.Newf(pgcode.FeatureNotSupported, "cannot use tenant names")
+	}
+
+	row, err := p.execCfg.InternalExecutor.QueryRowEx(
+		ctx, "get-tenant", p.Txn(), sessiondata.NodeUserSessionDataOverride,
+		`SELECT info FROM system.tenants WHERE name = $1`, tenantName,
+	)
+	if err != nil {
+		return nil, err
+	} else if row == nil {
+		return nil, pgerror.Newf(pgcode.UndefinedObject, "tenant %q does not exist", tenantName)
+	}
+
+	info := &descpb.TenantInfo{}
+	infoBytes := []byte(tree.MustBeDBytes(row[0]))
+	if err = protoutil.Unmarshal(infoBytes, info); err != nil {
```

Comment:
[nit you should probably just ignore and I probably shouldn't be typing] I think one of the reasons we use this `if err ...` construction is to restrict the scope of the `err` variable. But if we are assigning to an existing err variable, that kinda defeats the point.